src/tike/ptycho/solvers/combined.py

```diff
@@ -64,19 +63,48 @@ def combined(
     return {'psi': psi, 'probe': probe, 'cost': cost, 'scan': scan}
 
 
-def update_probe(op, pool, data, psi, scan, probe, num_iter=1):
+def _compute_intensity(op, psi, scan, probe):
+    farplane = op.fwd(
+        psi=psi,
+        scan=scan,
+        probe=probe,
+    )
+    return op.xp.sum(
+        op.xp.square(op.xp.abs(farplane)),
+        axis=(2, 3),
+    ), farplane
+
+
+def _update_probe(op, pool, data, psi, scan, probe, num_iter=1):
     """Solve the probe recovery problem."""
 
-    # TODO: Cache object patche between mode updates
+    # TODO: Cache object patches between mode updates
+    intensity = [
+        _compute_intensity(op, psi, scan, probe[..., m:m + 1, :, :])[0]
+        for m in range(probe.shape[-3])
+    ]
+    intensity = op.xp.array(intensity)
+
```

Comment:
However, this code chunk is a patch to the conjugate-gradient solver which avoids reduces the number of FFTs by caching the detector intensity contribution from the probe modes that are not being updated.

Reply:
Does this mean reducing the number of FFT is a code optimization for both LS and CG solvers, regardless of the LS algorithm?

Reply:
The least-squares method intrinsically performs less FFTs per iteration when compared to the conjugate gradient method. This PR also contains a patch which reduces the number of FFTs performed by the conjugate gradient method.